src/enrichment_tables/memory/config.rs

```
@@ -43,7 +48,34 @@ pub struct MemoryConfig {
     /// By default, there is no size limit.
     #[serde(skip_serializing_if = "vector_lib::serde::is_default")]
     pub max_byte_size: Option<u64>,
-
+    /// The namespace to use for logs. This overrides the global setting.
```

Let's extract all source related fields in a new struct.

Groups:
* common fields (here)
* source related fields (new struct)
* sink related fields? (another struct)

Maybe we also need builds validation checks so that bad configs fail early (didn't look deep into this, assuming that some fields don't make sense if the table is used as source/sink)

---

I would say that we keep sink related fields here too, because they are pretty much required. The component is not usable without these.

I agree with source related fields, it will be clearer if they are separate.